libraries/listen.rb

```diff
@@ -34,8 +34,8 @@ def self.converted_listen_ports_and_addresses(node)
       return [] unless node['apache']['listen_ports'] || node['apache']['listen_addresses']
       Chef::Log.warn "node['apache']['listen_ports'] and node['apache']['listen_addresses'] are deprecated in favor of node['apache']['listen']. Please adjust your cookbooks"
 
-      node['apache']['listen_addresses'].uniq.each_with_object([]) do |address, listen|
-        node['apache']['listen_ports'].uniq.each do |port|
+      (node['apache']['listen_addresses'] || []).uniq.each_with_object([]) do |address, listen|
+        (node['apache']['listen_ports'] || []).uniq.each do |port|
```

Comment:
We shouldn't have to define `apache.listen_ports`, as we get that from `apache.listen`. We could set it for tests as you say, but there should be no need to set it in the actual cookbook, IMHO.